include/small/region.h

```diff
@@ -372,16 +414,36 @@ region_aligned_alloc_xc_cb(void *ctx, size_t size)
 struct RegionGuard {
 	struct region *region;
 	size_t used;
+	bool armed;
 
 	RegionGuard(struct region *region_arg)
 		: region(region_arg),
-		  used(region_used(region_arg))
+		  used(region_used(region_arg)),
+		  armed(true)
         {
 		/* nothing */
 	}
 
-	~RegionGuard() {
-		region_truncate(region, used);
+	/** Call truncate before exiting scope. */
+	void trigger()
+	{
+		if (armed) {
+			region_truncate(region, used);
+			armed = false;
+		}
+	}
+
+	/** Reuse existing guard after instead of adding new one. */
+	void arm()
```

Comment:
I don't understand why you need this change - it isn't used anywhere.

Reply:
After offline discussion decided to keep guard as it is.